prov/efa/src/efa_user_info.c

```diff
@@ -302,28 +304,42 @@ bool efa_user_info_should_support_hmem(int version, const struct fi_info *hints)
 	 */
 	if (FI_VERSION_GE(version, FI_VERSION(1, 18)) && hmem_ops[FI_HMEM_CUDA].initialized) {
 		EFA_INFO(FI_LOG_CORE,
-			"User is using API version >= 1.18. CUDA library and device are availble, claim support of FI_HMEM.");
+			"User is using API version >= 1.18. CUDA library and "
+			"device are availble, claim support of FI_HMEM.");
 		return true;
 	}
 
 	if (ofi_hmem_p2p_disabled()) {
 		EFA_WARN(FI_LOG_CORE,
 			"FI_HMEM capability currently requires peer to peer "
-			"support, which is disabled because FI_HMEM_P2P_DISABLED was set to 1/on/true.");
+			"support, which is disabled because "
+			"FI_HMEM_P2P_DISABLED was set to 1/on/true.");
 		return false;
 	}
 
-	if (!efa_device_support_rdma_read()) {
+	if (!FI_VERSION_GE(version, FI_VERSION(1, 18))) {
+		extra_version_info = " the application is using libfabric "
+			"API <1.18, so CUDA cannot be used.";
+	}
```

Comment:
It seems that the structure can be 

```
if (!FI_VERSION_GE(version, FI_VERSION(1, 18))) {
   extra_info = "xxx"
} else {
     assert(!hmem_ops[FI_HMEM_CUDA].initialized);
   extra_info = "xxx"
}
```

Reply:
> I was trying to leave room for the possibility that both of those checks would fail, in which case I would like to capture that in the log.

I think it is possible for both to fail. But if you want to capture both in log,  you will need to put an "and" between the two sentence. I am not sure it worth the added complexity

Reply:
re-worked.  Please take a look and see what you think.